esphome/components/lc709203f/lc709203f.cpp

```diff
@@ -0,0 +1,302 @@
+        this->status_clear_warning();
+      }
+    }
+  } else if (this->state_ == LC709203F_STATE_INIT) {
+    // We should only get here if the init sequence failed during the setup() function.
+    //  This would likely occur because of a repeated failure on the I2C bus.
+    if (this->set_register_(LC709203F_IC_POWER_MODE, LC709203F_POWER_MODE_ON) == i2c::NO_ERROR) {
+      if (this->set_register_(LC709203F_APA, this->apa_) == i2c::NO_ERROR) {
+        if (this->set_register_(LC709203F_CHANGE_OF_THE_PARAMETER, this->pack_voltage_) == i2c::NO_ERROR) {
+          this->state_ = LC709203F_STATE_RSOC;
+        }
+      }
+    }
+  } else if (this->state_ == LC709203F_STATE_RSOC) {
+    // We implement a delay here to send the initial RSOC command.
+    //  This should run once on the first update() after initialization.
+    if (this->set_register_(LC709203F_INITIAL_RSOC, 0xAA55) == i2c::NO_ERROR) {
+      this->state_ = LC709203F_STATE_TEMP_SETUP;
+    }
+  } else if (this->state_ == LC709203F_STATE_TEMP_SETUP) {
+    // This should run once on the second update() after initialization.
+    if (this->temperature_sensor_ != nullptr) {
+      // This assumes that a thermistor is attached to the device as shown in the datahseet.
+      if (this->set_register_(LC709203F_STATUS_BIT, 0x0001) == i2c::NO_ERROR) {
+        if (this->set_register_(LC709203F_THERMISTOR_B, this->b_constant_) == i2c::NO_ERROR) {
+          this->state_ = LC709203F_STATE_NORMAL;
+        }
+      }
+    } else {
+      // The device expects to get updates to the temperature in this mode.
+      //  I am not doing that now. The temperature register defaults to 25C.
+      //  In theory, we could have another temperature sensor and have ESPHome
+      //  send updated temperature to the device occasionally, but I have no idea
+      //  how to make that happen.
+      if (this->set_register_(LC709203F_STATUS_BIT, 0x0000) == i2c::NO_ERROR) {
+        this->state_ = LC709203F_STATE_NORMAL;
+      }
+    }
+  }
+}
+
+void Lc709203f::dump_config() {
+  ESP_LOGCONFIG(TAG, "LC709203F:");
+  LOG_I2C_DEVICE(this);
+  if (this->is_failed()) {
+    // Nothing in this code actually sets the device to failed, so this should never trigger.
+    //  I am leaving this in incase I want to implement a check during init to verify the
+    //  correct device is on the bus.
+    ESP_LOGCONFIG(TAG, "  I2C Device at 0x0B does not appear to be a LC709203F");
+  }
+  LOG_UPDATE_INTERVAL(this);
+  ESP_LOGCONFIG(TAG, "  Pack Size: %d mAH", this->pack_size_);
+  ESP_LOGCONFIG(TAG, "  Pack APA: 0x%02X", this->apa_);
+  if (this->pack_voltage_ == 0x0000) {
+    ESP_LOGCONFIG(TAG, "  Pack Rated Voltage: 3.8V");
+  } else {
+    // This is only true if the pack_voltage_ is 0x0001. The config validator should
+    //  have already made sure that this value can be only 0 or 1.
+    ESP_LOGCONFIG(TAG, "  Pack Rated Voltage: 3.7V");
+  }
```

Comment:
Instead of storing two copies of the same string:
```suggestion
  // This is only true if the pack_voltage_ is 0x0001. The config validator should
  //  have already made sure that this value can be only 0 or 1.
  ESP_LOGCONFIG(TAG, "  Pack Rated Voltage: 3.%sV", this->pack_voltage_ == 0x0000 ? "8" : "7");
```